spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/BaseSparkAction.java

```diff
@@ -200,6 +203,18 @@ protected Dataset<FileInfo> manifestListDS(Table table, Set<Long> snapshotIds) {
     return toFileInfoDS(manifestLists, MANIFEST_LIST);
   }
 
+  protected Dataset<FileInfo> statisticsFileDS(Table table, Set<Long> snapshotIds) {
+    Predicate<StatisticsFile> predicate;
+    if (snapshotIds == null) {
```

Comment:
I see that `manifestListLocations` accepts a set of snapshot IDs and constructs a predicate internally. Do we anticipate arbitrary predicates in the future? If we think that users would most likely just want to find statistics files for given snapshots, then we can build the predicate internally.

Just asking.

Reply:
I originally coded it to be the same style as  `manifestListLocations`. Then @jackye1995 gave a comment to change this to accept predicate. I felt the general predicate is nice. So, I updated it. 

If the user wants to filter based on path or blob metadata, no change is required in the future if we support the predicate.

Reply:
Sounds good!